src/lib/settings/settings.py

```diff
@@ -0,0 +1,280 @@
+        ]
+
+    def get_monitoring_groups(self, resources: List[str]) -> List[str]:
+        """Get monitoring groups by resources list."""
+        matched_groups = set()  # Prevent duplicates
+
+        for group in self.monitoring_groups.get("monitoring_groups", []):
+            resource_groups = []
+            for monitored_resource in SettingConfigs.RESOURCE_TYPES:
+                resource_groups += group.get(monitored_resource, [])
+
+            for resource in resources:
+                matched_groups.update(
+                    group["group_name"]
+                    for res in resource_groups
+                    # TODO: replace fnmatch with == when _process_monitoring_groups() implemented
+                    if res["name"] and fnmatch(resource, res.get("name"))
+                )
+
+        return list(matched_groups)
+
+    def get_recipients(
+        self, monitoring_groups: List[str], notification_type: str
+    ) -> List[dict]:
+        """Get recipients by monitoring groups."""
+        matched_recipients = []
+
+        for recipient in self.recipients.get("recipients", []):
+            subscriptions = recipient.get("subscriptions", [])
+            for subscription in subscriptions:
+                for monitoring_group in monitoring_groups:
+                    if subscription.get("monitoring_group") == monitoring_group:
+                        if (
+                            notification_type == "alert" and subscription.get("alerts")
+                        ) or (
+                            notification_type == "digest" and subscription.get("digest")
+                        ):
+                            recipient_info = {
+                                "recipient": recipient.get("recipient"),
+                                "delivery_method": recipient.get("delivery_method"),
+                            }
+                            if recipient_info not in matched_recipients:
+                                matched_recipients.append(recipient_info)
+
+        return matched_recipients
+
+    # Methods for settings_validator (work with raw settings)
+    def get_monitored_environment_names_raw_gs(self) -> List[str]:
+        """Retrieves the monitored_environment names from the General settings (raw)."""
+        return [
+            m_env["name"]
+            for m_env in self._raw_settings[SettingFileNames.GENERAL].get(
+                "monitored_environments", []
+            )
+        ]
+
+    def get_monitored_account_id_and_region_raw_gs(self) -> List[str]:
+        """Retrieves the monitored environments 'account_id|region' from the General settings (raw)."""
+        return [
+            f"""{m_env["account_id"]}|{m_env["region"]}"""
+            for m_env in self._raw_settings[SettingFileNames.GENERAL].get(
+                "monitored_environments", []
+            )
+        ]
+
+    def get_delivery_method_names_raw_gs(self) -> List[str]:
```

Comment:
Maybe, it's better not to have any get method which rely on _raw_settings.

So far, we don't have "processed" for monitoring groups, but we can pretend "raw" state = "processed" stage for now. -> we can make these "get_" methods final, accessing processed settings.
Later - implement this raw -> processed transformations is the only change needed